### PR TITLE
CEDS-2962 Reorder MRN Entry

### PR DESCRIPTION
**Pull request by janeturner**
Reorder journey so that MRN Entry is after Start page. 
New journey is:
1. /start
2. /mrn-entry
3. /contact-details
4. /how-many-files-upload

### CODE REVIEW

File: test/controllers/ContactDetailsControllerSpec.scala [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 package controllers
 
-import controllers.actions.DataRetrievalAction
+import controllers.actions.{DataRetrievalAction, MrnRequiredActionImpl}
```
Comment (udinobi) [inline test/controllers/ContactDetailsControllerSpec.scala:19]: Uhm. I would not go with the implementation class, MrnRequiredActionImpl, but with the trait.
Check the 2 comments below

File: test/controllers/ContactDetailsControllerSpec.scala [hunk 2]
```diff
@@ -34,13 +34,20 @@ class ContactDetailsControllerSpec extends ControllerSpecBase {
...
 
   def controller(signedInUser: SignedInUser, eori: String, dataRetrieval: DataRetrievalAction = new FakeDataRetrievalAction(None)) =
-    new ContactDetailsController(new FakeAuthAction(signedInUser), new FakeEORIAction(eori), dataRetrieval, mockAnswersConnector, mcc, page)(
-      mcc.executionContext
-    )
+    new ContactDetailsController(
+      new FakeAuthAction(signedInUser),
+      new FakeEORIAction(eori),
+      dataRetrieval,
+      new MrnRequiredActionImpl(mcc),
```
Comment (udinobi) [inline test/controllers/ContactDetailsControllerSpec.scala:46]:
```
inject[MrnRequiredAction]
```

File: test/controllers/ContactDetailsControllerSpec.scala (hunk 1 above)
Comment (janeturner) [reply test/controllers/ContactDetailsControllerSpec.scala]: Fair enough, I had just looked at how it was done in `HowManyFilesUploadControllerSpec`, I've updated it now.

File: test/controllers/ContactDetailsControllerSpec.scala [hunk 3]
```diff
@@ -16,31 +16,40 @@
...
 
   def controller(signedInUser: SignedInUser, eori: String, dataRetrieval: DataRetrievalAction = new FakeDataRetrievalAction(None)) =
-    new ContactDetailsController(new FakeAuthAction(signedInUser), new FakeEORIAction(eori), dataRetrieval, mockAnswersConnector, mcc, page)(
-      mcc.executionContext
-    )
+    new ContactDetailsController(
+      new FakeAuthAction(signedInUser),
+      new FakeEORIAction(eori),
+      dataRetrieval,
+      inject[MrnRequiredAction],
```
Comment (glidester) [inline test/controllers/ContactDetailsControllerSpec.scala:48]: Why not create a 'FakeMrnRequiredAction' like the other two actions? Would save you having to extend GuiceOneAppPerSuite and Injecting in the test class. Also keep consistency of approach :slightly_smiling_face:

File: app/models/UserAnswers.scala [hunk 4]
```diff
@@ -22,8 +22,8 @@ import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 
 case class UserAnswers(
   eori: String,
-  contactDetails: Option[ContactDetails] = None,
   mrn: Option[MRN] = None,
+  contactDetails: Option[ContactDetails] = None,
```
Comment (MaciejRewera) [inline app/models/UserAnswers.scala:26]: Nit: Don't think this was necessary, but nice touch ;-)

File: conf/app.routes [hunk 5]
```diff
@@ -10,12 +10,12 @@ GET         /upload-error                       controllers.ErrorPageController.
...
 
-GET         /contact-details                    controllers.ContactDetailsController.onPageLoad
-POST        /contact-details                    controllers.ContactDetailsController.onSubmit
-
 GET         /mrn-entry                          controllers.MrnEntryController.onPageLoad
 POST        /mrn-entry                          controllers.MrnEntryController.onSubmit
 
+GET         /contact-details                    controllers.ContactDetailsController.onPageLoad
+POST        /contact-details                    controllers.ContactDetailsController.onSubmit
+
```
Comment (MaciejRewera) [inline conf/app.routes:18]: 👍

File: test/controllers/ContactDetailsControllerSpec.scala (hunk 3 above)
Comment (MaciejRewera) [reply test/controllers/ContactDetailsControllerSpec.scala:48]: Agree with @glidester . What is even more important, is that in order for injector to work, you need to start the application andthis takes significantly more time. Also, it is arguable if such test is a Unit Test anymore.

File: app/models/UserAnswers.scala (hunk 4 above)
Comment (janeturner) [reply app/models/UserAnswers.scala:26]: No I didn't think so, but I wanted to keep things neat :P

File: test/controllers/ContactDetailsControllerSpec.scala (hunk 3 above)
Comment (janeturner) [reply test/controllers/ContactDetailsControllerSpec.scala]: Thanks, I've updated this now

**Review by glidester — APPROVED** [review]

**Pull request merged**